sled-agent/src/services.rs

```diff
@@ -4203,12 +4225,17 @@ impl ServiceManager {
             // we can safely send a message on it to say the sled has
             // been synchronized.
             let queue = self.metrics_queue();
-            if !queue.notify_time_synced_sled(self.sled_id()).await {
-                error!(
+            match queue.notify_time_synced_sled(self.sled_id()) {
+                Ok(_) => debug!(
                     self.inner.log,
-                    "Failed to notify metrics queue of sled \
-                     time synchronization, metrics may not be produced."
-                );
+                    "Notified metrics task that time is now synced",
+                ),
+                Err(e) => error!(
+                    self.inner.log,
+                    "Failed to notify metrics task that \
+                     time is now synced, metrics may not be produced.";
```

Comment:
How bad is this failure? It seems pretty bad at a glance since `on_time_sync()` can only fire once. Maybe worth filing an issue that we should retry this or wait for it or something?

Reply:
It's not great, since we'll never produce data from this sled. It also Should Not Happen, since we're not starting many zones at once before time is synchronized. I've filed #8022 to track this, and referenced it in 34822583d.